### PR TITLE
:tada: article on Argo gateway

### PR DESCRIPTION
**Pull request by hlminh2000**

### CODE REVIEW

File: blog/rosi_bajari/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+title: Rosi Posts
+category: rosi_bajari
+permalink: /blog/category/rosi_bajari
+header: 
+    version: small
+    title: Technical Business Analyst
+    image: header-logo-crop.png
+    page: team
+    icon: icon-blog
+---
```
Comment (hlminh2000) [inline blog/rosi_bajari/index.md:12]: @rosibaj I added this for you :) it'll populate this page: https://softeng.oicr.on.ca/blog/category/rosi_bajari

Hope that works :)

Comment (rosibaj) [reply blog/rosi_bajari/index.md:12]: thanks minh thats prefect!

File: _posts/minh_ha/2020-08-15-Argo-gateway.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+So we searched for ways to remove the burden of managing back-end data from the UI, and arrived here:
+
+<image src="{{ site.urlimg }}/minh_ha/argo_gateway/with_gateway.png" />
+
+This would not have been immediately clear for us as a path forward, without some past experience with GraphQL. GraphQL was a technology the team had been experimenting with, but it never came to be seen as a central piece of our tech stack. Some of the services built in KF were using GraphQL, but we made a big mistake of treating GraphQL as if it was REST. They were individual services with GraphQL interfaces, but it was left to the consumer (i.e the UI) to coordinate between them. Multiple GraphQL endpoint also necessarily resulted in multiple GraphQL client instances, and/or none at all. The n+1 problem that GraphQL is known for solving still remained in our experiments, and other benefits we saw did not justify the learning required. But seeing where things went wrong, we decided to give the technology another try with Argo.
+
```
Comment (rosibaj) [inline _posts/minh_ha/2020-08-15-Argo-gateway.md:69]: ARGO all caps - its an acronym!

File: _posts/minh_ha/2020-08-15-Argo-gateway.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+Ultimately this removed the need to manage back-end data in the UI as application state entirely. We were no longer faced with the dilemma between global vs local state. Data requirements of each component could be collocated with the rendering logic without risking big refactors down the road. Data update can happen automatically without any boiler plate code to write. We get the best of both worlds when it comes to developer experience. Once we mentally think of the Apollo cache store for what it is (a dumb cache store, rather than a smart application state store), then backend data was no longer something our application code had to manage.
```
Comment (rosibaj) [inline _posts/minh_ha/2020-08-15-Argo-gateway.md]: - Data update`s` can happen automatically
-

File: _posts/minh_ha/2020-08-15-Argo-gateway.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+However, when this is not the case, Redux is still a very attractive option. Imagine, a test-taking program that needs to mostly run off-line and transmit test results only when there is network availability... or a survey app for collecting data in the fields where no network is available... or maybe even the app I am using to write this post right now (VScode), etc... could be benefiting from a strong application state management system like Redux. (Although it seems Apollo client is becoming a full-blown state management solution lately... I have not yet really experienced that approach yet, would love to hear if you have)..
+
+Finally, this is not to claim that all of the improved frontend developer experience in Argo has been due to the Gateway and Apollo. Argo was started at the prime time of React with the new react-hooks that drastically changed how we think about React components entirely. Other technologies that we adopted also took some time to mature and for us to learn the usage that works for us. These include Typescript, Next.JS, Storybook, etc... We arrived at our current stack by taking risks on new technologies, making mistakes and learning from them.
```
Comment (rosibaj) [inline _posts/minh_ha/2020-08-15-Argo-gateway.md]: ARGO

**Comment by justincorrigible** [comment]
Could not delete the added DS_Store file here, so I've opened a PR on your PR to do so.

image

**Review by justincorrigible — APPROVED** [review]
:shipit: Good to go!

**Pull request merged**